### PR TITLE
Added 5.12.0 changelog

### PR DESCRIPTION
**Pull request by Blue0ctober**
## Description

Adds 5.12.0 changelog

## Motivation and Context

 5.12.0 Release!

### CODE REVIEW

File: content/sensu-go/5.12/release-notes.md [hunk 1]
```diff
@@ -36,6 +36,51 @@ Read the [upgrade guide][1] for information on upgrading to the latest version o
...
+Added sensuctl dump command to output resources to a file or STDOUT
+Added agent --allow-list to list allowed executables for checks and check hooks.
+
+**IMPROVEMENTS:**
+
+Licensed feature: Namespaces listed in both the web UI and sensuctl are now limited to those which the user has access to
+The event.check.name field has been added as a supported field selector
+Both the API and sensuctl can now be used to delete Assets
+The use of Protobuf serialization/deserialization over websocket can now be negotiated between agent and backend.
+We fixed some bugs in watchers through refactoring -- what does it mean to end users?
```
Comment (echlebek) [inline content/sensu-go/5.12/release-notes.md]: Users who noticed checks failing to update or schedule should see an improvement.

File: content/sensu-go/5.12/release-notes.md [hunk 2]
```diff
@@ -36,6 +36,51 @@ Read the [upgrade guide][1] for information on upgrading to the latest version o
...
+The generic etcd watcher now keeps track of revisions.
```
Comment (echlebek) [inline content/sensu-go/5.12/release-notes.md]: Probably best to not mention this.

**Review by echlebek — COMMENTED** [review]
You can also let users know to expect performance improvements in the web UI for depoyments with a lot of events and entities.

File: content/sensu-go/5.12/release-notes.md (hunk 2 above)
Comment (cwjohnston) [inline content/sensu-go/5.12/release-notes.md]:
```suggestion
```

File: content/sensu-go/5.12/release-notes.md (hunk 1 above)
Comment (cwjohnston) [inline content/sensu-go/5.12/release-notes.md]: We'll move this down to fixes. Suggest we replace it with something to cover Eric's comments about web UI speed improvements.

```suggestion
Web UI performance is improved UI for deployments with lots of events and entities.
```

File: content/sensu-go/5.12/release-notes.md [hunk 3]
```diff
@@ -36,6 +36,51 @@ Read the [upgrade guide][1] for information on upgrading to the latest version o
...
+To prevent writing sensitive data to logs, decoded check result and keepalive payloads are no longer logged by the backend.
+
+**FIXES:**
+
+Tabular display of filters via Sensuctl now displays && or || as appropriate for inclusive and exclusive filters, respectively.
+Requesting events from the GET /events/:entity API endpoint now returns events only for the specified entity
+Running sensuctl config view without configuration no longer causes a crash
+Creating an entity via sensuctl with the --interactive flag now prompts for the entity name when not provided on the command line.
+Check hooks with stdin: true now receive actual event data on STDIN instead of an empty event.
+
```
Comment (cwjohnston) [inline content/sensu-go/5.12/release-notes.md:78]:
```suggestion
Some issues with check scheduling and updating have been fixed by refactoring the backend's watcher implementation.

```

File: content/sensu-go/5.12/release-notes.md [hunk 4]
```diff
@@ -36,6 +36,51 @@ Read the [upgrade guide][1] for information on upgrading to the latest version o
...
+**KNOWN ISSUES:**
+
+Deleting an asset via the API or sensuctl will only delete the asset resource within Sensu backend.
```
Comment (cwjohnston) [inline content/sensu-go/5.12/release-notes.md]: I think this seems redundant with the next line so I suggest striking it

```suggestion
```

File: content/sensu-go/5.12/release-notes.md [hunk 5]
```diff
@@ -36,6 +36,51 @@ Read the [upgrade guide][1] for information on upgrading to the latest version o
...
+
+**NEW FEATURES:**
+
+Added sensu-agent and sensuctl binary builds for FreeBSD 
+Added sensuctl dump command to output resources to a file or STDOUT
+Added agent --allow-list to list allowed executables for checks and check hooks.
+
+**IMPROVEMENTS:**
+
+Licensed feature: Namespaces listed in both the web UI and sensuctl are now limited to those which the user has access to
```
Comment (cwjohnston) [inline content/sensu-go/5.12/release-notes.md]:
```suggestion
Assets now support defining multiple builds, reducing the number of individual assets needed to cover disparate platforms in your infrastructure.
Licensed feature: Namespaces listed in both the web UI and sensuctl are now limited to those which the user has access to.
```

**Review by cwjohnston — CHANGES_REQUESTED** [review]

File: content/sensu-go/5.12/release-notes.md (hunk 2 above)
Comment (Blue0ctober) [reply content/sensu-go/5.12/release-notes.md]:
```suggestion
```

File: content/sensu-go/5.12/release-notes.md [hunk 6]
```diff
@@ -36,6 +36,51 @@ Read the [upgrade guide][1] for information on upgrading to the latest version o
...
+Added agent --allow-list to list allowed executables for checks and check hooks.
+
+**IMPROVEMENTS:**
+
+Assets now support defining multiple builds, reducing the number of individual assets needed to cover disparate platforms in your infrastructure.
+Licensed feature: Namespaces listed in both the web UI and sensuctl are now limited to those which the user has access to.
+The event.check.name field has been added as a supported field selector
+Both the API and sensuctl can now be used to delete Assets
+The use of Protobuf serialization/deserialization over websocket can now be negotiated between agent and backend.
+Web UI performance is improved UI for deployments with lots of events and entities.
```
Comment (echlebek) [inline content/sensu-go/5.12/release-notes.md]: The English is a bit English in this sentence :joy:

Comment (cwjohnston) [inline content/sensu-go/5.12/release-notes.md]:
```suggestion
Web UI performance is improved for deployments with many events and entities.
```

File: content/sensu-go/5.12/release-notes.md [hunk 7]
```diff
@@ -36,6 +36,51 @@ Read the [upgrade guide][1] for information on upgrading to the latest version o
...
+The resource caches can now rebuild themselves in case of failures.
```
Comment (cwjohnston) [inline content/sensu-go/5.12/release-notes.md:62]: @echlebek is this a user-facing improvement?

File: content/sensu-go/5.12/release-notes.md [hunk 8]
```diff
@@ -36,6 +36,51 @@ Read the [upgrade guide][1] for information on upgrading to the latest version o
...
+**IMPORTANT:**
+
+Due to changes in the release process, Sensu binary-only archives are now named following the pattern `sensu-go_5.12.0_$OS_$ARCH.tar.gz`, where $OS is the operating system name and $ARCH is the CPU architecture. These archives include all files in the top level directory. See the [installation guide][16] for the latest download links.
+Line item
+
+**NEW FEATURES:**
+
+Added sensu-agent and sensuctl binary builds for FreeBSD 
+Added sensuctl dump command to output resources to a file or STDOUT
+Added agent --allow-list to list allowed executables for checks and check hooks.
```
Comment (cwjohnston) [inline content/sensu-go/5.12/release-notes.md]:
```suggestion
Agents can now be configured with a list executables which check and check hook commands are allowed to run.
```

File: content/sensu-go/5.12/release-notes.md [hunk 9]
```diff
@@ -36,6 +36,51 @@ Read the [upgrade guide][1] for information on upgrading to the latest version o
...
+
+**IMPORTANT:**
+
+Due to changes in the release process, Sensu binary-only archives are now named following the pattern `sensu-go_5.12.0_$OS_$ARCH.tar.gz`, where $OS is the operating system name and $ARCH is the CPU architecture. These archives include all files in the top level directory. See the [installation guide][16] for the latest download links.
+Line item
+
+**NEW FEATURES:**
+
+Added sensu-agent and sensuctl binary builds for FreeBSD 
+Added sensuctl dump command to output resources to a file or STDOUT
```
Comment (cwjohnston) [inline content/sensu-go/5.12/release-notes.md]:
```suggestion
Added sensuctl dump command to output resources to a file or STDOUT, making it easier to backup your Sensu backends.
```

File: content/sensu-go/5.12/release-notes.md [hunk 10]
```diff
@@ -36,6 +36,51 @@ Read the [upgrade guide][1] for information on upgrading to the latest version o
...
+**August 22, 2019** &mdash; The latest release of Sensu Go, version 5.12.0, is now available for download. There are some exciting feature additions in this release including the ability to output resources to a file from sensuctl and more granularity of control around checks/check hooks execution with an agent allow list. Additionally, this release includes the capability to delete assets and some more stability fixes around watcher functionality. See the [upgrading guide]() to upgrade Sensu to version 5.12.0.
+
+**IMPORTANT:**
+
+Due to changes in the release process, Sensu binary-only archives are now named following the pattern `sensu-go_5.12.0_$OS_$ARCH.tar.gz`, where $OS is the operating system name and $ARCH is the CPU architecture. These archives include all files in the top level directory. See the [installation guide][16] for the latest download links.
+Line item
+
+**NEW FEATURES:**
+
+Added sensu-agent and sensuctl binary builds for FreeBSD
```
Comment (cwjohnston) [inline content/sensu-go/5.12/release-notes.md]:
```suggestion
Operators can now authenticate to Sensu via OpenID Direct Connect (OIDC) using sensuctl, see our [authentication documentation][17] for details.
Added sensu-agent and sensuctl binary builds for FreeBSD on amd64 and 386 architectures.
```

File: content/sensu-go/5.12/release-notes.md [hunk 11]
```diff
@@ -36,6 +36,51 @@ Read the [upgrade guide][1] for information on upgrading to the latest version o
...
+
+**SECURITY:**
+
+To prevent writing sensitive data to logs, decoded check result and keepalive payloads are no longer logged by the backend.
+
+**FIXES:**
+
+Tabular display of filters via Sensuctl now displays && or || as appropriate for inclusive and exclusive filters, respectively.
+Requesting events from the GET /events/:entity API endpoint now returns events only for the specified entity
+Running sensuctl config view without configuration no longer causes a crash
```
Comment (cwjohnston) [inline content/sensu-go/5.12/release-notes.md]:
```suggestion
Running sensuctl config view without configuration no longer causes a crash.
```

File: content/sensu-go/5.12/release-notes.md [hunk 12]
```diff
@@ -36,6 +36,51 @@ Read the [upgrade guide][1] for information on upgrading to the latest version o
...
+Event and Entity resources can now be created via the API using the POST verb
+
+**SECURITY:**
+
+To prevent writing sensitive data to logs, decoded check result and keepalive payloads are no longer logged by the backend.
+
+**FIXES:**
+
+Tabular display of filters via Sensuctl now displays && or || as appropriate for inclusive and exclusive filters, respectively.
+Requesting events from the GET /events/:entity API endpoint now returns events only for the specified entity
```
Comment (cwjohnston) [inline content/sensu-go/5.12/release-notes.md]:
```suggestion
Requesting events from the GET /events/:entity API endpoint now returns events only for the specified entity.
```

File: content/sensu-go/5.12/release-notes.md [hunk 13]
```diff
@@ -36,6 +36,51 @@ Read the [upgrade guide][1] for information on upgrading to the latest version o
...
+
+Assets now support defining multiple builds, reducing the number of individual assets needed to cover disparate platforms in your infrastructure.
+Licensed feature: Namespaces listed in both the web UI and sensuctl are now limited to those which the user has access to.
+The event.check.name field has been added as a supported field selector
+Both the API and sensuctl can now be used to delete Assets
+The use of Protobuf serialization/deserialization over websocket can now be negotiated between agent and backend.
+Web UI performance is improved UI for deployments with lots of events and entities.
+The resource caches can now rebuild themselves in case of failures.
+Event and Entity resources can now be created via the API without an explicit namespace; the system will refer to the namespace in the request URL.
+Event and Entity resources can now be created via the API using the POST verb
```
Comment (cwjohnston) [inline content/sensu-go/5.12/release-notes.md]:
```suggestion
Event and Entity resources can now be created via the API using the POST verb.
```

File: content/sensu-go/5.12/release-notes.md [hunk 14]
```diff
@@ -36,6 +36,51 @@ Read the [upgrade guide][1] for information on upgrading to the latest version o
...
+Added sensu-agent and sensuctl binary builds for FreeBSD 
+Added sensuctl dump command to output resources to a file or STDOUT
+Added agent --allow-list to list allowed executables for checks and check hooks.
+
+**IMPROVEMENTS:**
+
+Assets now support defining multiple builds, reducing the number of individual assets needed to cover disparate platforms in your infrastructure.
+Licensed feature: Namespaces listed in both the web UI and sensuctl are now limited to those which the user has access to.
+The event.check.name field has been added as a supported field selector
+Both the API and sensuctl can now be used to delete Assets
```
Comment (cwjohnston) [inline content/sensu-go/5.12/release-notes.md]:
```suggestion
Both the API and sensuctl can now be used to delete Assets.
```

File: content/sensu-go/5.12/release-notes.md [hunk 15]
```diff
@@ -36,6 +36,51 @@ Read the [upgrade guide][1] for information on upgrading to the latest version o
...
+Tabular display of filters via Sensuctl now displays && or || as appropriate for inclusive and exclusive filters, respectively.
+Requesting events from the GET /events/:entity API endpoint now returns events only for the specified entity
+Running sensuctl config view without configuration no longer causes a crash
+Creating an entity via sensuctl with the --interactive flag now prompts for the entity name when not provided on the command line.
+Check hooks with stdin: true now receive actual event data on STDIN instead of an empty event.
+Some issues with check scheduling and updating have been fixed by refactoring the backend's watcher implementation.
+
+**KNOWN ISSUES:**
+
+Deleting an asset will not remove references to said asset. It is the operators responsibility to remove the asset from the runtime_assets field of the check, hook, filter, mutator, or handler.
```
Comment (cwjohnston) [inline content/sensu-go/5.12/release-notes.md:82]:
```suggestion
Authentication via OIDC is not yet supported in the Web UI.
Deleting an asset will not remove references to said asset. It is the operators responsibility to remove the asset from the runtime_assets field of the check, hook, filter, mutator, or handler.
```

File: content/sensu-go/5.12/release-notes.md (hunk 10 above)
Comment (cwjohnston) [reply content/sensu-go/5.12/release-notes.md]: TODO: add a link for authentication documentation at the end of the doc.

**Review by cwjohnston — CHANGES_REQUESTED** [review]

File: content/sensu-go/5.12/release-notes.md (hunk 7 above)
Comment (echlebek) [reply content/sensu-go/5.12/release-notes.md:62]: Users with large inventories should notice an improvement.

**Review by echlebek — APPROVED** [review]
LGTM aside from Cameron's comments

File: content/sensu-go/5.12/release-notes.md (hunk 7 above)
Comment (cwjohnston) [reply content/sensu-go/5.12/release-notes.md:62]: @echlebek are these caches on the backend, agent or both?

Comment (echlebek) [reply content/sensu-go/5.12/release-notes.md:62]: They are on the backend. This is an implementation detail, it doesn't necessarily need to be communicated to users.

**Pull request merged**